src/main/java/uk/gov/hmcts/reform/idam/web/strategic/PolicyService.java

```diff
@@ -0,0 +1,96 @@
+
+import static java.util.Collections.singletonList;
+import static java.util.Optional.ofNullable;
+
+@Slf4j
+@Service
+public class PolicyService {
+
+    public static final String ERROR_POLICY_CHECK_EXCEPTION = "Policy check exception.";
+
+    private final RestTemplate restTemplate;
+
+    private final ConfigurationProperties configurationProperties;
+
+    @Autowired
+    public PolicyService(RestTemplate restTemplate, ConfigurationProperties configurationProperties) {
+        this.restTemplate = restTemplate;
+        this.configurationProperties = configurationProperties;
+    }
+
+    /**
+     * @should return true when all actions return true
+     * @should return false when any action returns false
+     * @should throw exception when response is not successful
+     * @should return true when no actions are returned
+     * @should return true when actions is null
+     */
+    public boolean evaluatePoliciesForUser(final String uri, final String cookie, final String ipAddress) {
+        final String applicationName = configurationProperties.getStrategic().getPolicies().getApplicationName();
+
+        final String userSsoToken = StringUtils.substringAfter(cookie, "=");
+
+        final EvaluatePoliciesRequest request = new EvaluatePoliciesRequest()
+            .resources(singletonList(uri))
+            .application(applicationName)
+            .subject(new Subject().ssoToken(userSsoToken))
+            .environment(ImmutableMap.of("requestIp", singletonList(ipAddress)));
+
+        final HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.put(HttpHeaders.COOKIE, Collections.singletonList(cookie));
+        headers.setBearerAuth(userSsoToken);
+
+        final HttpEntity<EvaluatePoliciesRequest> httpEntity = new HttpEntity<>(request, headers);
+        final String url = String.format("%s/%s",
+            configurationProperties.getStrategic().getService().getUrl(),
+            configurationProperties.getStrategic().getEndpoint().getEvaluatePolicies());
+
+        final ResponseEntity<EvaluatePoliciesResponse> response = restTemplate
+            .exchange(url, HttpMethod.POST, httpEntity, EvaluatePoliciesResponse.class);
+
+        if (!response.getStatusCode().is2xxSuccessful()) {
+            throw new HttpClientErrorException(response.getStatusCode(), ERROR_POLICY_CHECK_EXCEPTION);
+        }
+
+        final boolean result = evaluateResult(response);
+        return result;
+    }
+
+    private boolean evaluateResult(ResponseEntity<EvaluatePoliciesResponse> response) {
+        final EvaluatePoliciesResponse result = ofNullable(response.getBody())
+            .orElse(new EvaluatePoliciesResponse());
+        for (EvaluatePoliciesResponseInner resultItem : result) {
+            final ActionMap actions = ofNullable(resultItem.getActions()).orElse(new ActionMap());
+            final boolean block = actions.values().stream()
```

Comment:
I really think we should have some logging somewhere about which policy the user failed - I guess that could be added in the API though.

Reply:
in api we would just have an SSO token, it wouldn't identify the user
at least here we have the username